### PR TITLE
Generate report zip

### PR DESCRIPTION
**Pull request by PhilipDeegan**
and dry run

Forgot to push last commit to PR 

## Summary by CodeRabbit

- **New Features**
	- Added a help message and command-line interface for simulation configuration.
	- Introduced build and runtime validation checks for PHARE configurations.
	- Enhanced reporting functionality to collect and organize PHARE reports into a zip file.
	- Implemented a new system for handling PHARE build configurations and environment settings.
- **Bug Fixes**
	- Fixed issue reporting instructions and known issues with OMPI symbol resolution and Python launching with C++ MPI init.
- **Documentation**
	- Updated instructions for reporting issues.
- **Refactor**
	- Updated project configuration to include the C language.
	- Improved handling of MPI library paths and parallelism checks.
	- Optimized cleanup logic in simulator tests for better synchronization.
- **Style**
	- Adjusted code formatting in the `SolverPPC` class.
- **Chores**
	- Updated `.gitignore` patterns and minimum required CMake versions for several projects.
- **Tests**
	- Modified test class inheritance and teardown procedures for improved validation testing.

### CODE REVIEW

**Comment by PhilipDeegan** [comment]
for #739

File: pyphare/pyphare_tests/pharein/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.3)
+cmake_minimum_required (VERSION 3.20.1)
```
Comment (PhilipDeegan) [inline pyphare/pyphare_tests/pharein/CMakeLists.txt:1]: TC was showing the following text

```
CMake Deprecation Warning at pyphare/pyphare_tests/test_pharesee/CMakeLists.txt:1 (cmake_minimum_required):
```

so no point in not updating these three files to be consistent with the rest of the project

File: tools/config/cmake.sh [hunk 2]
```diff
@@ -6,12 +6,17 @@ CWD="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )" && cd "$CWD"
 ROOT_DIR=$(cd ../.. && pwd)
 [ -f "${ROOT_DIR}/res/cmake/def.cmake" ] || (echo "Corruption detected" && exit 1) # confirm correct directory
```
Comment (nicolasaunai) [inline tools/config/cmake.sh:7]: maybe we need an error msg bit more explanatory: "Corruption detected: missing "def.cmake""

File: src/core/def/phare_config.hpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#ifndef PHARE_CORE_DEF_PHARE_CONFIG_HPP
+#define PHARE_CORE_DEF_PHARE_CONFIG_HPP
+
+#include <string>
+
+
+#if __has_include("core/def/_gen_sys.hpp")
```
Comment (nicolasaunai) [inline src/core/def/phare_config.hpp:7]: few comments to help me re-understand all this more quickly next time.
// if __gen_sys.hpp  exists, it has been generated by the phare_configurator (config.py) 
// if it does not, the  build_config() version below reflects that phare_configurator is off

**Review by nicolasaunai — APPROVED** [review]
few questions/comments other all good thx

File: res/cmake/def.cmake [hunk 4]
```diff
@@ -11,6 +11,10 @@ endif() # clang
 set (PHARE_LINK_FLAGS )
 set (PHARE_BASE_LIBS )
 
+if(NOT DEFINED PHARE_MPIRUN_POSTFIX)
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply res/cmake/def.cmake:14]: I think I have a comment locally actually

the gist

if, say, during testing, you spawn 10 processes that are doing `mpirun -n 2 ....` 

only one will actually run, the rest are queued, I find this strange, as to me, they should be independent.

File: src/amr/CMakeLists.txt [hunk 5]
```diff
@@ -77,7 +77,7 @@ add_library(${PROJECT_NAME} ${SOURCES_INC} ${SOURCES_CPP})
 target_compile_options(${PROJECT_NAME} PRIVATE ${PHARE_WERROR_FLAGS})
 set_property(TARGET ${PROJECT_NAME} PROPERTY INTERPROCEDURAL_OPTIMIZATION ${PHARE_INTERPROCEDURAL_OPTIMIZATION})
 
-target_link_directories(${PROJECT_NAME} PUBLIC ${MPI_LIBRARY_PATH})
+target_link_directories(${PROJECT_NAME} PUBLIC MPI::MPI_C)
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply src/amr/CMakeLists.txt:80]: likely, I'm kinda torn, sometimes I think core shouldn't know anything about mpi

File: res/cmake/options.cmake [hunk 6]
```diff
@@ -92,8 +92,8 @@ endif()
 option(bench "Compile PHARE Benchmarks" OFF)
 
 
-# -Dphare_configurator=OFF
-option(phare_configurator "Best guess setup for compile" OFF)
+# -Dphare_configurator=ON
+option(phare_configurator "Best guess setup for compile" ON)
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply res/cmake/options.cmake:96]: maybe :)

File: tools/config/cmake.sh (hunk 2 above)
Comment (PhilipDeegan) [reply tools/config/cmake.sh:7]: more like "Code base in inconsisntent state - reacquire from source". ?

The reasoning for the specific `def.cmake` file just to be clear, is that the `def.cmake` is what checks if the `phare_configurator` is on, and if so, spawns the process calling this script. so if the def.cmake is not found, something is very wrong

File: res/cmake/def.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply res/cmake/def.cmake:14]: will probably rm this to reassess anyway, we should probably have some mechanism for supporting this tho I think

File: res/cmake/options.cmake (hunk 6 above)
[the remark this answers is not in the document]
Comment (nicolasaunai) [reply res/cmake/options.cmake:96]: 🤷

[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply res/cmake/options.cmake:96]: to revert

File: src/amr/CMakeLists.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply src/amr/CMakeLists.txt:80]: [I got rid of MPICXX in the views PR](https://github.com/PhilipDeegan/PHARE/blob/views/src/core/CMakeLists.txt#L66)

File: res/cmake/def.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply res/cmake/def.cmake]: move to configurator if we detect OMPI

**Review by nicolasaunai — APPROVED** [review]

File: pyphare/pyphare/core/__init__.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+"""
+
+import sys
+import dataclasses
+
+
+def disabled_for_testing():
+    # check if any module is loaded from PHARE tests directory
+    from pathlib import Path
+
```
Comment (nicolasaunai) [inline pyphare/pyphare/core/__init__.py:20]: 🤭

**Review by nicolasaunai — APPROVED** [review]

**Pull request merged**